pagopa/io-react-native-wallet
# feat: [SIW-1149] Add getAttestation

> [!NOTE]  
> Co-authored by @grausof and @gispada 






This PR adds the flow to obtain a WTE attestation by using the integrity services of the iOS/Android device.

#### List of Changes



- Updates the `getAttestation` function according to our internal [design review](https://pagopa.atlassian.net/wiki/spaces/SIW/pages/946274401/IO-WALLET-DR-0009+Wallet+Trust+Evidence);
- Updates the `IntegrityContext` interface by adding a `getHardwareSignatureWithAuthData` function which must return a `HardwareSignatureWithAuthData` containing the signature of `clientData` and the authenticator data; 
- Moves platform specific implementations related to the `IntegrityContext` in two different platform specific files `integrity.ios.ts` and `integrity.android.ts`;
- Updates the `yaml` response for the `/token` endpoint;
- Updates `io-react-native-integrity` package version.


#### Motivation and Context



More on the required changes is documented in the aforementioned design review. 


#### How Has This Been Tested?





- Run an instance of `apps/io-wallet-user-func` from [io-wallet](https://github.com/pagopa/io-wallet) with an appropriate `local.settings.json` (I can provide it privately);
- Run an instance of `dev-proxy` in this example folder by providing a `.env` file pointing at the running azure function of the previous point (`http://localhost:7071/api/v1` by default). I can provide the remaining variables privately);
- Run the example app on both Android and iOS by providing a `.env` file. The `WALLET_PROVIDER_BASE_URL` should point to the IP of your local machine URL where the `dev-proxy` is running (again, I can provide the remaining variables privately). Consequentially press every available button in the app and everything should work just fine.

#### Checklist:
